src/databricks/labs/ucx/hive_metastore/table_size.py

```diff
@@ -55,12 +60,20 @@ def _try_load(self) -> Iterable[TableSize]:
     def snapshot(self) -> list[TableSize]:
         """
         Takes a snapshot of tables in the specified catalog and database.
+        Return None if the table cannot be found anymore.
 
         Returns:
             list[Table]: A list of Table objects representing the snapshot of tables.
         """
         return self._snapshot(partial(self._try_load), partial(self._crawl))
 
-    def get_table_size(self, table_full_name: str) -> int:
+    def _get_safe_table_size(self, table_full_name: str) -> int | None:
         logger.debug(f"Evaluating {table_full_name} table size.")
-        return self._spark._jsparkSession.table(table_full_name).queryExecution().analyzed().stats().sizeInBytes()
+        try:
+            return self._spark._jsparkSession.table(table_full_name).queryExecution().analyzed().stats().sizeInBytes()
+        except NotFound as nf:
```

Comment:
Spark doesn't yet throw the NotFound, catching Exception is okay here

Reply:
we were frequently getting NotFound exceptions here in our integration tests since many tests can run in parallel and tables might get removed.

Reply:
Do you suggest any changes here?

Reply:
corrected